cluster/gce/util.sh

```diff
@@ -3473,17 +3505,18 @@ function kube-down() {
   detect-project
   detect-node-names # For INSTANCE_GROUPS and WINDOWS_INSTANCE_GROUPS
 
-  echo "Bringing down cluster"
+  echo 'Bringing down cluster'
   set +e  # Do not stop on error
 
-  if [[ "${KUBE_DELETE_NODES:-}" != "false" ]]; then
+  if [[ "${KUBE_DELETE_NODES:-}" != 'false' ]]; then
     # Get the name of the managed instance group template before we delete the
     # managed instance group. (The name of the managed instance group template may
     # change during a cluster upgrade.)
-    local templates=$(get-template "${PROJECT}")
+    local templates
+    templates=$(get-template "${PROJECT}")
 
-    local all_instance_groups=(${INSTANCE_GROUPS[@]:-} ${WINDOWS_INSTANCE_GROUPS[@]:-})
-    for group in ${all_instance_groups[@]:-}; do
+    local all_instance_groups=("${INSTANCE_GROUPS[@]:-}" "${WINDOWS_INSTANCE_GROUPS[@]:-}")
+    for group in "${all_instance_groups[@]:-}"; do
```

Comment:
are you sure we need this? I suspect we may want globbing

Reply:
For array entries where there is no whitespace, the outcome will be the same. But for entries where there is whitespace (although I guess there are no such groups in reality?), there is indeed a difference: I am under the impression that each array entry contains 1 group name (in which case globbing is not wanted), where a group name if it contains a whitespace with the current code on master will be split.

```
INSTANCE_GROUPS=('group 0')
INSTANCE_GROUPS+=('group 1')
WINDOWS_INSTANCE_GROUPS=('wingroup a')
WINDOWS_INSTANCE_GROUPS+=('wingroup b')

# Current code on master
all_instance_groups=(${INSTANCE_GROUPS[@]:-} ${WINDOWS_INSTANCE_GROUPS[@]:-})
for group in ${all_instance_groups[@]:-}; do
        echo "$group"
done

echo '**********'

# Updated code in my patch
new_all_instance_groups=("${INSTANCE_GROUPS[@]:-}" "${WINDOWS_INSTANCE_GROUPS[@]:-}")
for group in "${new_all_instance_groups[@]:-}"; do
        echo "$group"
done
```

The output would be

```
group
0
group
1
wingroup
a
wingroup
b
**********
group 0
group 1
wingroup a
wingroup b
```